### ISSUE TITLE
Relationship Graphs

### ISSUE THREAD

**Issue by cybershambles** [monicahq/monica#357]
Imagine I Added person X, added person Y, added Child Z.... I'd love to be able to link there people together automatically as people

then have them displayed as as graph. 

----
also, you cannot link existing people
![image](https://user-images.githubusercontent.com/7995198/27323187-0216941c-55e4-11e7-9fae-f1e28c1ea27c.png)

Doesn't seem to actually link the people.. just added the text field?
![image](https://user-images.githubusercontent.com/7995198/27323242-3f071496-55e4-11e7-9321-12e10d0f95c2.png)

**Comment by UltraPhil** [comment]
I don't think it would work very well to try to automatically match people together, but having the opportunity to add relationship informations between people (brothers, sisters, cousins, spouses, children) would be great!

**Comment by dancyfits** [comment]
This is the lowest hanging fruit in my opinion.  Having this feature would do the most in making Monica a complete product, for me.

**Comment by lathomas64** [comment]
yes was very dissapointed that I could not have a significant other listed as a separate person instead of basically a note on the person.

**Comment by sid3windr** [comment]
The linking of contacts is now implemented and working, this issue should be (and remain?) only for the graph part :) Would also love to see this!

**Comment by cybershambles** [comment]
:D

**Comment by djaiss** [comment]
Are you aware of a tool that would help me build those relationship graphs?

**Comment by dancyfits** [comment]
https://github.com/lazychaser/laravel-nestedset
I'm not up to date with the PHP scene, but that looks pretty helpful!  Maybe at least a good an example to work from

**Comment by turtles2** [comment]
@djaiss The graph part of this interests me. I am willing to find the tool and build them. I just need some guidance on what we are looking for visually.

**Comment by djaiss** [comment]
@dancyfits imho GraphQL is unrelated with the relationship graph concept. GraphQL is an alternative to traditional REST APIs, and the work to implement it in Monica would take months and months.

@turtles2 @dancyfits before finding out how it should look like, we should define what it would do, considering the current structure of the relationships between contacts. Also I need to be convinced that we need this feature, because I'm not sure what it would bring 😀

**Comment by turtles2** [comment]
@djaiss My interest in implementing this is due to the technical nature of it. I was hoping for guidance from you and others on what it should do precisely

**Comment by dancyfits** [comment]
@djaiss This type of problem is one of the main reasons GraphQL was built in the first place.
https://dev-blog.apollodata.com/explaining-graphql-connections-c48b7c3d6976#de1c

GraphQL is not an alternative for REST APIs.
http://graphql.org/blog/rest-api-graphql-wrapper/

**Comment by cybershambles** [comment]
I'd also like to be able to tag people into groups.. I have social circles.. like a workplace or hobby or group of friends.

**Comment by dancyfits** [comment]
@cybershambles That you can do.  On a contact's page, "Edit tag" under their name.

**Comment by brortao** [comment]
Agree that GraphQL is not relevant to this issue -- it's a tool for building APIs, but this issue is about visualizing a graph.

I've used this front-end graph framework in the past, and it's pretty solid and well-maintained: http://js.cytoscape.org/

If Monica already has API endpoints to get all contacts and all relationships, then visualizing such a graph isn't a very large task. Do these endpoints exist?

Just visualizing the social graph is the MVP, but another neat feature could be e.g. dragging a line between people on the graph, then getting a small popup to select the relationship type. This'd be a fast way to enter relationships!

**Comment by vergissberlin** [comment]
Here is another library which could help to implement it:
https://fperucic.github.io/treant-js/

**Comment by MrModest** [comment]
Maybe this script will be somehow helpful in implementing a built-in feature: https://github.com/jmcker/monica2neo4j

**Comment by rayandaod** [comment]
Is there an update on this? I think this could be very helpful
